storage/innobase/handler/handler0alter.cc

```diff
@@ -5072,6 +5081,10 @@ prepare_inplace_alter_table_dict(
 	}
 
 	if (ctx->need_rebuild() && ctx->new_table->supports_instant()) {
+		if (ha_alter_info->create_info->vers_info
+			.with_system_versioning)
+			goto not_instant_add_column;
+
```

Comment:
@kevgs, please comment.

Reply:
Add versioning is:
- add `sys_trx_start`, `sys_trx_end` fields (with values of `CUR_TRX_ID` and `MAX_INT64` respectively)
- add `DICT_TF2_VERSIONED`
- set up `dict_table_t::vers_row_start`, `dict_table_t::vers_row_end`

As I understand this can be performed instantly. So it's a missing feature.

Reply:
> Which forms of ALTER TABLE do you think are inherently incompatible with versioning?

Sorry I didn't get this question. Incompatible in what way?

Reply:
https://github.com/tempesta-tech/mariadb/commit/5f0f7cdd43498975a76f576d5c3dfd1d8e4d6be5

Reply:
Thank you for the clarifying comment.
As far as I understand, it explains why adding system versioning cannot be done instantly. (It cannot be done instantly anyway, because it would involve DROP PRIMARY KEY, ADD PRIMARY KEY.)
The comment does not clarify why instant ADD COLUMN is not allowed on a table that contains system versioning. Please see my follow-up comment in https://github.com/tempesta-tech/mariadb/commit/5f0f7cdd43498975a76f576d5c3dfd1d8e4d6be5.